packages/amplication-server/src/core/resource/resourceBtm.service.ts

```diff
@@ -0,0 +1,261 @@
+    MultiLineText: "string",
+    Email: "string",
+    WholeNumber: "int",
+    DateTime: "datetime",
+    DecimalNumber: "float",
+    Lookup: "enum",
+    MultiSelectOptionSet: "enum",
+    OptionSet: "enum",
+    Boolean: "bool",
+    GeographicLocation: "string",
+    Id: "int",
+    CreatedAt: "datetime",
+    UpdatedAt: "datetime",
+    Roles: "string",
+    Username: "string",
+    Password: "string",
+    Json: "string",
+  };
+
+  constructor(
+    private readonly gptService: GptService,
+    private readonly prisma: PrismaService,
+    private readonly userActionService: UserActionService
+  ) {}
+
+  async triggerBreakServiceIntoMicroservices({
+    resourceId,
+    userId,
+  }: {
+    resourceId: string;
+    userId: string;
+  }): Promise<UserAction> {
+    const resource = await this.getResourceDataForBtm(resourceId);
+
+    const prompt = this.generatePromptForBreakTheMonolith(resource);
+
+    const conversationParams = [
+      {
+        name: "userInput",
+        value: prompt,
+      },
+    ];
+
+    const userAction = await this.gptService.startConversation(
+      ConversationTypeKey.BreakTheMonolith,
+      conversationParams,
+      userId,
+      resourceId
+    );
+
+    return userAction;
+  }
+
+  async finalizeBreakServiceIntoMicroservices(
+    userActionId: string
+  ): Promise<BreakServiceToMicroserviceResult> {
+    const { resourceId, metadata } = await this.userActionService.findOne({
+      where: {
+        id: userActionId,
+      },
+    });
+
+    const userActionStatus = await this.userActionService.calcUserActionStatus(
+      userActionId
+    );
+
+    if (userActionStatus !== EnumUserActionStatus.Completed) {
+      return {
+        status: EnumUserActionStatus[userActionStatus],
+        originalResourceId: resourceId,
+        data: null,
+      };
+    }
+
+    const recommendations = await this.preparePromptResultToBtmRecommendation(
+      JSON.stringify(metadata),
+      resourceId
+    );
+
+    return {
+      status: EnumUserActionStatus.Completed,
+      originalResourceId: resourceId,
+      data: recommendations,
+    };
+  }
+
+  generatePromptForBreakTheMonolith(resource: ResourcePartial): string {
+    const entityIdNameMap = resource.entities.reduce((acc, entity) => {
+      acc[entity.id] = entity.name;
+      return acc;
+    });
+
+    const prompt: BreakTheMonolithPromptInput = {
+      dataModels: resource.entities.map((entity) => {
+        return {
+          name: entity.name,
+          fields: entity.versions[0].fields.map((field) => {
+            return {
+              name: field.name,
+              dataType:
+                field.dataType == EnumDataType.Lookup
+                  ? entityIdNameMap[field.properties["relatedEntityId"]]
+                  : this.dataTypeMap[field.dataType],
+            };
+          }),
+        };
+      }),
+    };
+
+    return JSON.stringify(prompt);
+  }
+
+  async preparePromptResultToBtmRecommendation(
```

Comment:
This is not PROMPT - this is the results
```suggestion
  async prepareBtmRecommendations(
```